py/legacyanalysis/sersic-agn.py

```diff
@@ -0,0 +1,269 @@
+    
+        b = Duck()
+        b.ra = ra
+        b.dec = dec
+    
+        W,H = 80,80
+        wcs = wcs_for_brick(b, W=W, H=H)
+        targetrd = np.array([wcs.pixelxy2radec(x,y) for x,y in
+                             [(1,1),(W,1),(W,H),(1,H),(1,1)]])
+        ccds = survey.ccds_touching_wcs(wcs)
+        print(len(ccds), 'CCDs')
+    
+        ims = [survey.get_image_object(ccd) for ccd in ccds]
+        keepims = []
+        for im in ims:
+            h,w = im.shape
+            if h >= H and w >= W:
+                keepims.append(im)
+        ims = keepims
+        gims = [im for im in ims if im.band == 'g']
+        rims = [im for im in ims if im.band == 'r']
+        zims = [im for im in ims if im.band == 'z']
+        nk = min([len(gims), len(rims), len(zims), 5])
+        ims = gims[:nk] + rims[:nk] + zims[:nk]
+        print('Keeping', len(ims), 'images')
+        
+        tims = [im.get_tractor_image(pixPsf=True, hybridPsf=True, normalizePsf=True, splinesky=True, radecpoly=targetrd) for im in ims]
+    
+        bands = 'grz'
+    
+        devsrc = DevGalaxy(RaDecPos(ra,dec), NanoMaggies(**dict([(b,10.) for b in bands])), EllipseESoft(0., 0., 0.))
+        tr = Tractor(tims, [devsrc])
+        tr.freezeParam('images')
+        tr.optimize_loop()
+        print('Fit DeV source:', devsrc)
+        devmods = list(tr.getModelImages())
+        showboth(tims, devmods, wcs, bands);
+        s = flux_string(devsrc.brightness)
+        plt.suptitle('DeV model: ' + s + '\ndchisq 0.')
+        plt.savefig('src%02i-dev.png' % isrc)
+        devchi = 2. * tr.getLogLikelihood()
+        
+        dasrc = DevAgnGalaxy(devsrc.pos.copy(), devsrc.brightness.copy(), devsrc.shape.copy(), NanoMaggies(**dict([(b,1.) for b in bands])))
+        #dasrc = DevAgnGalaxy(RaDecPos(ra,dec), NanoMaggies(**dict([(b,10.) for b in bands])), EllipseESoft(0., 0., 0.), NanoMaggies(**dict([(b,1.) for b in bands])))
+        tr = Tractor(tims, [dasrc])
+        tr.freezeParam('images')
+        tr.optimize_loop()
+        print('Fit Dev+PSF source:', dasrc)
+        damods = list(tr.getModelImages())
+        showboth(tims, damods, wcs, bands)
+        s1 = flux_string(dasrc.brightnessDev)
+        s2 = flux_string(dasrc.brightnessPsf)
+        pcts = [100. * dasrc.brightnessPsf.getFlux(b) / dasrc.brightnessDev.getFlux(b) for b in bands]
+        s3 = ', '.join(['%.2f' % p for p in pcts])
+        dachi = 2. * tr.getLogLikelihood()
+        plt.suptitle('DeV + Point Source model: DeV %s, PSF %s' % (s1, s2) + ' (%s %%)' % s3 +
+                     '\ndchisq %.1f' % (dachi - devchi))
+        plt.savefig('src%02i-devcore.png' % isrc)
+    
+        #sersrc = SersicGalaxy(RaDecPos(ra, dec), NanoMaggies(**dict([(b,10.) for b in bands])), EllipseESoft(0.5, 0., 0.), SersicIndex(4.0))
+        sersrc = SersicGalaxy(devsrc.pos.copy(), devsrc.brightness.copy(), devsrc.shape.copy(), SersicIndex(4.0))
+        tr = Tractor(tims, [sersrc])
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['Tractor' may be undefined, or defined from star imports: legacypipe.survey, tractor, tractor.devagn, tractor.galaxy](https://app.codacy.com/app/dstndstn/legacypipe/pullRequest?prid=4730106)